overlord/snapstate/snapmgr.go

```diff
@@ -680,7 +680,7 @@ func Manager(st *state.State, runner *state.TaskRunner) (*SnapManager, error) {
 	// FIXME: drop the task entirely after a while
 	runner.AddHandler("clear-aliases", func(*state.Task, *tomb.Tomb) error { return nil }, nil)
 	runner.AddHandler("set-auto-aliases", m.doSetAutoAliases, m.undoRefreshAliases)
-	runner.AddHandler("setup-aliases", m.doSetupAliases, m.undoSetupAliases)
+	runner.AddHandler("setup-aliases", m.doSetupAliases, m.undoRefreshAliases)
```

Comment:
we need a test that the undoing the set-auto-aliases setup-aliases sequence does the right thing, I'm not quite sure that this will do the right thing as the point of this undo the set-auto-aliases itself is not undone yet

Reply:
to be clear the test should test a situation where there is an interesting set of things to undo and written such that it can be run on master and here and pass both places

Reply:
Updated, thanks! I also added tests for the rest of the PR and will move it out of draft.

Reply:
out of all added tests `TestDoUndoSetupAliasesAutoPruneOldAliases` and `TestDoUndoSetupAliasesAutoPruneOldAliasesConflict` are mainly focusing on tasks interactions, they should work on master if the specific checks are removed, I added those for more confidence.